CRFModuleValidation/CRFMotionRecorder.swift

```diff
@@ -209,83 +214,85 @@ public class CRFMotionRecorder : RSDSampleRecorder {
     }
 }
 
-/// `CRFAccelerometerRecord` is intended to be used for recording raw accelerometer data.
-public struct CRFAccelerometerRecord: RSDSampleRecord {
+public struct CRFMotionRecord : RSDSampleRecord {
     
     public let uptime: TimeInterval
     public let timestamp: TimeInterval?
     public let stepPath: String
     public let timestampDate: Date?
-    public let sensorType: CRFMotionRecorderType
+    
+    public let sensorType: CRFMotionRecorderType?
+    public let eventAccuracy: Int?
+    public let referenceCoordinate: String?
+    public let heading: Double?
     
     public let x: Double?
     public let y: Double?
     public let z: Double?
+    public let w: Double?
     
-    public init(startUptime: TimeInterval, stepPath: String, data: CMAccelerometerData) {
+    public init?(startUptime: TimeInterval, stepPath: String, data: CMDeviceMotion, sensorType: CRFMotionRecorderType) {
+
+        var eventAccuracy: Int?
+        var referenceCoordinate: String?
+        var w: Double?
+        
+        switch sensorType {
+        case .attitude:
+            self.x = data.attitude.quaternion.x
+            self.y = data.attitude.quaternion.y
+            self.z = data.attitude.quaternion.z
+            w = data.attitude.quaternion.w
```

Comment:
unclear why you're not assigning this directly to self.w...?

Reply:
b/c otherwise I have to assign it for every case.  In a Swift struct initializer you can only assign a value once, but you *have* to assign all values (even `nil`).

Reply:
oh right, makes sense